File: EjemplosJson/Python/CRUDPythonStudents/Student_controller.py
from StudentCrud_model import StudentCrud
from Student_model import Student

class StudentController:
    def __init__(self):
        self.student_crud = StudentCrud()

    def add_student(self, student_name, grades_list):
        student_id = self.student_crud.generate_student_id()
        student_object = Student(student_id, student_name, grades_list)
        self.student_crud.add_student(student_object)
        return student_object

    def get_all_students(self):
        return self.student_crud.get_all_students()

    def edit_student(self, student_id, new_student_name, new_grades_list):
        return self.student_crud.edit_student(student_id, new_student_name, new_grades_list)

    def delete_student(self, student_id):
        return self.student_crud.delete_student(student_id)

File: EjemplosJson/JavaScript/CRUDJavaScriptStudents/Controller/StudentController.js
const StudentCrud = require("../Model/StudentCrud");
const Student = require("../Model/Student");

class StudentController {
    constructor() {
        this.studentCrud = new StudentCrud();
    }

    addStudent(studentName, gradesList) {
        const newStudentId = this.studentCrud.generateStudentId();
        const newStudent = new Student(newStudentId, studentName, gradesList);
        this.studentCrud.addStudent(newStudent);
        return newStudent;
    }

    getAllStudents() {
        return this.studentCrud.getAllStudents();
    }

    editStudent(studentId, newStudentName, newGradesList) {
        return this.studentCrud.editStudent(studentId, newStudentName, newGradesList);
    }

    deleteStudent(studentId) {
        return this.studentCrud.deleteStudent(studentId);
    }
}

module.exports = StudentController;
